cleanlab_studio/studio/studio.py

```diff
@@ -358,3 +361,61 @@ def poll_cleanset_status(self, cleanset_id: str, timeout: Optional[int] = None)
 
         except (TimeoutError, CleansetError):
             return False
+
+    def autofix_dataset(
+        self,
+        original_df: pd.DataFrame,
+        cleanset_id: str,
+        params: Optional[Dict[str, Union[int, float]]] = None,
+        strategy="optimized_training_data",
+    ) -> pd.DataFrame:
+        """
+        This method returns the auto-fixed dataset. It works for text or tabular dataset only.
+        Args:
+            cleanset_id (str): ID of cleanset.
+            original_df (pd.DataFrame): The original dataset in DataFrame format.
+            params (dict, optional): Default parameter dictionary containing confidence threshold for auto-relabelling, and
+                fraction of rows to drop for each issue type. If not provided, default values will be used.
+                This dictionary includes the following options:
+
+                    * drop_ambiguous (float): Fraction of rows to drop when encountering ambiguous data. Default is 0.0 (no rows dropped).
+                    * drop_label_issue (float): Fraction of rows to drop when facing label-related issues. Default is 0.5 (50% of rows dropped).
+                    * drop_near_duplicate (float): Fraction of rows to drop for near-duplicate data. Default is 0.5 (50% of rows dropped).
+                    * drop_outlier (float): Fraction of rows to drop for outlier data. Default is 0.2 (20% of rows dropped).
+                    * relabel_confidence_threshold (float): Confidence threshold for auto-relabelling. Default is 0.95.
+                For example, the default values are:
+                {
+                    'drop_ambiguous': 0.0,
+                    'drop_label_issue': 0.5,
+                    'drop_near_duplicate': 0.5,
+                    'drop_outlier': 0.2,
+                    'relabel_confidence_threshold': 0.95
+                }
+
+            Specify values in params to customize the behavior for specific scenarios. If params are provided, the values in params take precedence over default ones.
+            
+            strategy (str): Auto-fixing strategy to use,
+                Possible strategies: optimized_training_data, drop_all_issues, suggested_actions
```

Comment:
```suggestion
        Improves a dataset by applying automatically-suggested corrections based on issues detected by Cleanlab.
        Args:
            cleanset_id (str): ID of the cleanset from the Project for this Dataset.
            original_df (pd.DataFrame): The original dataset (must be a DataFrame, so only text and tabular datasets are currently supported).
            params (dict, optional): Optional parameters to control how many data points from each type of detected data issue are auto-corrected or filtered (prioritizing the more severe instances of each issue). If not provided, default `params` values will be used.
                The `params` dictionary includes the following options:

                    * drop_ambiguous (float): Fraction of the data points detected as ambiguous to exclude from the dataset.
                    * drop_label_issue (float): Fraction of the data points with label issues to exclude from the dataset.
                    * drop_near_duplicate (float): Fraction of the data points detected as near duplicates to exclude from the dataset.
                    * drop_outlier (float): Fraction of the data points detected as outliers to exclude from the dataset.
                    * relabel_confidence_threshold (float): Confidence threshold for the suggested label, data points with label issues that also exceed this threshold are re-labeled as the suggested label.

            strategy (str): What strategy to use for auto-fixing the dataset out of the following possibilities: 
            ['optimized_training_data', 'drop_all_issues', 'suggested_actions'].
            Each of these possibilities corresponds to a default setting of the `params` dictionary, designed to be used in different scenarios. If specified, the `params` argument will override this argument. Specify 'optimized_training_data' when your goal is to auto-fix training data to achieve the best ML performance on randomly split test data. Specify 'drop_all_issues' to instead exclude all datapoints detected to have issues from the dataset. Specify 'suggested_actions' to instead apply the suggested action to each data point that is displayed in the Web Application (e.g. relabeling for label issues, dropping for outliers, etc).
```

Reply:
@aditya1503 once you add this in, I think this PR should be good for the Studio team to take over.

Please share the rest of your benchmarking results on other tabular & image datasets with me!

Reply:
yep I'll share